docs/DeveloperGuide.md

```diff
@@ -1,33 +1,297 @@
+running AthletiCLI instance.
+
+**Step 2 - Activity Parsing:** The Parser parses the raw input to obtain the arguments of the activity. Given that all 
+parameters are provided correctly and no exception is thrown, a new activity object is created.
+
+**Step 3 - Command Parsing:** In addition the parser will create an `AddActivityCommand` object with the newly added 
+activity attached to it. The command implements the `AddActivityCommand#execute()` operation and is passed to 
+the AthletiCLI instance.
+
+**Step 4 - Activity Addition:** The AthletiCLI instance executes the `AddActivityCommand` object. The command will 
+access the data and retrieve the currently stored list of activities stored inside it. The new `Activity` object is 
+added to the list.
+
+**Step 5 - User Interaction:** Once the activity is successfully added, a confirmation message is displayed to the user.
+
+The following sequence diagram shows how the `add-activity` operation works:
+<p  align="center" >
+  <img width="100%" src="images/AddActivity.svg" alt="Sequence Diagram of add-activity"/>
+</p>
+
+#### [Implemented] Tracking activity goals
+
+With the `set-activity-goal` feature, users can set periodic goals for their activities.
+The fulfillment of these goals is tracked automatically and can be evaluated by the user at any time.
+
+These are the key components and their roles in the architecture of the goal tracking:
+* `SetActivityGoalCommand`: encapsulates the execution of the `set-activity-goal` command. It adds 
+  the activity goal to the data.
+* `ActivityGoal`: represents the activity goal that is to be added and contains functionality to 
+  track the fulfillment of the goal. 
+* `ActivityList`: contains key functionality to retrieve and filter the activity list according to the specified 
+  properties of the goal.
+
+Given below is an example usage scenario and how the goal setting and tracking mechanism behaves at 
+each step.
+
+1. **Step 1 - Input Capture:** The user issues a `set-activity-goal ...` which is captured and passed to the 
+   Parser by the running AthletiCLI instance.
+2. **Step 2 - Goal Parsing:** The Parser parses the raw input to obtain the sports, target and timespan of the goal. 
+   Given that all these parameters are provided correctly and no exception is thrown, a new activity goal object is 
+   created.
+3. **Step 3 - Command Parsing:** In addition the parser will create a `SetActivityGoalCommand` object with the newly 
+   added activity goal attached to it. The command implements the `SetActivityGoalCommand#execute()` operation and is 
+   passed to the AthletiCLI instance.
+4. **Step 4 - Goal Addition:** The AthletiCLI instance executes the `SetActivityGoalCommand` object. The command will 
+   access the data and retrieve the currently stored list of activity goals stored inside it. The new `ActivityGoal` 
+   object is added to the list.
```

Comment:
Good format! Labelling each of the steps makes it easy to understand what is being done in each step